ResearchKit/Common/ORKAnswerFormat_Internal.h

```diff
@@ -135,6 +135,10 @@ ORK_DESIGNATE_CODING_AND_SERIALIZATION_INITIALIZERS(ORKTimeIntervalAnswerFormat)
 - (UIImage *)minimumImage;
 - (NSArray<ORKTextChoice *> *)textChoices;
 
+@optional
```

Comment:
I see.

Then how about doing the inherited `ORKTextScaleAnswerFormatProvider` protocol anyway, moving `textChoices` to the inherited protocol, and then replacing the recurrent `[_formatProvider textChoices]` test by `[[_formatProvider class] conformsToProtocol:@protocol(ORKTextScaleAnswerFormatProvider)]` (or a `-(BOOL)isTextFormatProvider:(id<ORKScaleAnswerFormatProvider>formatProvider` helper method doing that), and then casting the format provider to `(id<ORKTextScaleAnswerFormatProvider)` to use those exclusive methods if the test succeeds. 

It's more boilerplatery but it feels safer and intent may be clearer.